django/contrib/auth/models.py

```diff
@@ -157,6 +157,32 @@ def create_superuser(self, username, email=None, password=None, **extra_fields):
 
         return self._create_user(username, email, password, **extra_fields)
 
+    def with_perm(self, perm, is_active=True, include_superusers=True, backend=None, obj=None):
+        if backend is None:
+            backends = auth._get_backends(return_tuples=True)
+            if len(backends) == 1:
+                backend, _ = backends[0]
+            else:
+                raise ValueError(
+                    'You have multiple authentication backends configured and '
+                    'therefore must provide the `backend` argument.'
+                )
+        elif not isinstance(backend, str):
+            raise TypeError(
+                'backend must be a dotted import path string (got %r).'
+                % backend
+            )
+        else:
+            backend = auth.load_backend(backend)
+        if hasattr(backend, 'with_perm'):
```

Comment:
Do we want to (or perhaps rather, can we) add `BaseBackend.with_perm()`? It could allow us to avoid `hasattr()`, although custom backends may need updating.

Reply:
I don't that we should do this. `BaseBackend` contains only basic methods.

Reply:
... also we cannot use `User` in the `BaseBackend` so it will be hard to return something consistent.

Reply:
Cool. Just thought the question should be asked.